protocol/communities/manager.go

```diff
@@ -2119,25 +2119,17 @@ func (m *Manager) DeclineRequestToJoin(dbRequest *RequestToJoin) (*Community, er
 	return community, nil
 }
 
-func (m *Manager) isUserRejectedFromCommunity(signer *ecdsa.PublicKey, community *Community, requestClock uint64) (bool, error) {
-	declinedRequestsToJoin, err := m.persistence.DeclinedRequestsToJoinForCommunity(community.ID())
+func (m *Manager) shouldUserRetainDeclined(signer *ecdsa.PublicKey, community *Community, requestClock uint64) (bool, error) {
```

Comment:
I believe the name is more precise than before. However, I'm open to discussion. Could you please suggest any alternatives?

Reply:
how about: `IsRequestToJoinOutdated` or `IsRequestToJoinExpired`

Reply:
How about:
- IsApprovalDelayedAfterDecline
- IsNotYetApprovableAfterDecline
- IsApprovalBlockedAfterDecline
- IsApprovalNotAllowedAfterDecline

If change returns a result from false to true it can be renamed to
- IsEligibleForApprovalAfterDecline
- CanApproveAfterDecline
- IsApprovalAllowedAfterDeclineTimeout
- HasDeclineCooldownElapsed
- IsApprovalPermittedAfterDecline

Reply:
Thank you for the suggestions. I would prefer to stick with shouldUserRetainDeclined if you don't mind, as it conveys the same meaning as the proposed names.